pkg/products/marin3r/rateLimitService.go

```diff
@@ -381,3 +402,119 @@ func GetRateLimitFromConfig(c *corev1.ConfigMap) (*yamlRateLimit, error) {
 	}
 	return ratelimitconfig.Descriptors[0].RateLimit, nil
 }
+
+func (r *RateLimitServiceReconciler) getConfigMapConfiguration(ctx context.Context, client k8sclient.Client, installation *integreatlyv1alpha1.RHMI) (*yamlRoot, error) {
+	var stagingconfig yamlRoot
+
+	if !integreatlyv1alpha1.IsRHOAMMultitenant(integreatlyv1alpha1.InstallationType(integreatlyv1alpha1.InstallationType(installation.Spec.Type))) {
+		stagingconfig = yamlRoot{
+			Domain: "apicast-ratelimit",
+			Descriptors: []yamlDescriptor{
+				{
+					Key:   genericKey,
+					Value: "slowpath",
+					RateLimit: &yamlRateLimit{
+						Unit:            r.RateLimitConfig.Unit,
+						RequestsPerUnit: r.RateLimitConfig.RequestsPerUnit,
+					},
+				},
+			},
+		}
+	} else {
+		limitPerTenant, err := r.getLimitPerTenantFromConfigMap(client, ctx)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get tenant limit from a config map: %v", err)
+		}
+
+		stagingconfig = yamlRoot{
+			Domain: "apicast-ratelimit",
+			Descriptors: []yamlDescriptor{
+				{
+					Key:   genericKey,
+					Value: "slowpath",
+					RateLimit: &yamlRateLimit{
+						Unit:            r.RateLimitConfig.Unit,
+						RequestsPerUnit: r.RateLimitConfig.RequestsPerUnit,
+					},
+				},
+				{
+					Key:   headerMatch,
+					Value: "per-mt-limit",
+					Descriptors: []yamlDescriptor{
+						{
+							Key: headerKey,
+							RateLimit: &yamlRateLimit{
+								Unit:            r.RateLimitConfig.Unit,
+								RequestsPerUnit: limitPerTenant,
+							},
+						},
+					},
+				},
+			},
+		}
+	}
+	return &stagingconfig, nil
+}
+
+func (r *RateLimitServiceReconciler) getLimitPerTenantFromConfigMap(client k8sclient.Client, ctx context.Context) (uint32, error) {
+
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      multitenantLimitConfigMap,
+			Namespace: r.Namespace,
+		},
+	}
+
+	err := client.Get(context.TODO(), types.NamespacedName{Namespace: r.Namespace, Name: multitenantLimitConfigMap}, configMap)
+	if err != nil {
+		if k8sError.IsNotFound(err) {
+			_, err := controllerutil.CreateOrUpdate(ctx, client, configMap, func() error {
+				if configMap.Data == nil {
+					configMap.Data = map[string]string{}
+				}
+				configMap.Data[multitenantRateLimit] = fmt.Sprint(r.getLimitPerTenant())
+
+				return nil
+			})
+			if err != nil {
+				return 0, err
+			}
+		} else {
+			return 0, fmt.Errorf("Error when getting the config map %w", err)
+		}
+	}
```

Comment:
This could be refactored to avoid many if else statements and improve the readability of the code 
```suggestion
	if err != nil && !k8sError.IsNotFound(err) {
	    return 0, fmt.Errorf("Error when getting the config map %w", err)
	} else if k8sError.IsNotFound(err) {
	    configMap.Data = map[string]string{}
	} 
	
	_, err := controllerutil.CreateOrUpdate(ctx, client, configMap, func() error {
		configMap.Data[multitenantRateLimit] = fmt.Sprint(r.getLimitPerTenant())
		return nil
	})
```

Reply:
Thanks a lot @jjaferson , I've updated the code but with a slight change. I have moved the createOrUpdate inside the `else if`, I don't think we should be reconciling the config map as we might want this to be fully configurable by us? 
So my idea here is, create config map if it doesn't exists and set the value of rate limit to `quota per minute / 3k` with 1 being the minimum. If a config map exists, we don't want to reconcile it as we wan't to be able to update it (as long as we won't have a new quota I think we will want to set it manually to something more than 1) WDYT?